RadixWallet/Clients/HTTPClient/HTTPClient+Interface.swift

```diff
@@ -5,5 +5,5 @@ public struct HTTPClient: Sendable, DependencyKey {
 
 // MARK: HTTPClient.ExecuteRequest
 extension HTTPClient {
-	public typealias ExecuteRequest = @Sendable (URLRequest) async throws -> Data
+	public typealias ExecuteRequest = @Sendable (_ request: URLRequest, _ isStatusCodeValid: ((Int) -> Bool)?) async throws -> Data
```

Comment:
instead of a closure to determine whether a status code is considered valid, I would have it receive an array of accepted status codes. We could use just `Int` or define an enum with the predefined values ([example](https://gist.github.com/ollieatkinson/322338df8a5220d649ac01ff11e7de12)).

Reply:
Either what @matiasbzurovski suggested, or return the status code along with the Data and let the consumer decide if it is valid or not.

Reply:
@matiasbzurovski not sure if we need such an ample enum, but an array of `Int` sounds good

Reply:
we may not want to add all the cases (since we are not mapping the int value to the corresponding case anyway), but if define at least the expected ones (`200` & `202` for now) we could refer it directly (`HttpStatusCode.ok` vs `BadHTTPResponseCode.expected`)

Reply:
agree